### PR TITLE
fix: stylelint to webpack error output

### PR DESCRIPTION
**Pull request by crash7**
Hi!
This is an attempt to fix https://github.com/JaKXz/stylelint-webpack-plugin/issues/102

- Remove chalk dep (not used anymore, stylelint already does the coloring)
- Add instances of Error to the compilation errors/warnings arrays
- Update tests

Closes https://github.com/JaKXz/stylelint-webpack-plugin/issues/102

### CODE REVIEW

File: lib/run-compilation.js [hunk 1]
```diff
@@ -41,12 +40,12 @@ module.exports = function runCompilation (options, compiler, done) {
 
   compiler.plugin('after-emit', function afterEmit (compilation, callback) {
     if (warnings.length) {
-      compilation.warnings.push(chalk.yellow(options.formatter(warnings)));
+      compilation.warnings.push(new Error(options.formatter(warnings)));
```
Comment (JaKXz) [inline lib/run-compilation.js]: Should warnings really be wrapped in `Error` objects too? Does this break builds if you only have warnings, but not errors?

Comment (crash7) [inline lib/run-compilation.js]: No, it doesn't break anything, the `Error` wrapper is because webpack is expecting the warnings/errors to be instance of `Error` in order to display them in the compilation errors

**Comment by crash7** [comment]
@JaKXz thanks for the review! :smile: 
re: yarn.lock, chalk is still a dependecy of stylelint itself, that's why the lock is still the same

File: lib/run-compilation.js (hunk 1 above)
Comment (JaKXz) [inline lib/run-compilation.js]: Interesting, good to know. And this behaviour is the same across v1, v2, and v3?

Comment (crash7) [inline lib/run-compilation.js]: I need to check v1, so far I tested v2 and v3.

**Comment by crash7** [comment]
@JaKXz after a lot of digging, I found that the problem is not related with this plugin nor webpack :tada: 

However I couldn't find the right way to create errors/warnings in the doc :disappointed:, I read some comments saying that it should be instances of `Error` and also checked what some loader plugins (eslint, babel) and found the same, but again, nothing in the docs. Probably because webpack itself supports both things: strings and `Error`: https://github.com/webpack/webpack/blob/master/lib/Stats.js#L162-L167

Well, turns out I'm also using [friendly-errors-webpack-plugin](https://github.com/geowarin/friendly-errors-webpack-plugin), which inside expects the errors to be an instance of `Error` [here](https://github.com/geowarin/friendly-errors-webpack-plugin/blob/master/src/formatters/defaultError.js#L12)

TL;DR: there isn't an error in the library, but this PR will make this plugin compatible with [friendly-errors-webpack-plugin](https://github.com/geowarin/friendly-errors-webpack-plugin)

**Comment by JaKXz** [comment]
@crash7 okay that makes sense. I guess I'll have to release this as a feature instead of a patch fix then, since it is new behaviour. Great digging!
